src/composeappengine.cc

```diff
@@ -42,46 +45,44 @@ void ComposeAppEngine::remove(const App& app) {
 }
 
 bool ComposeAppEngine::isRunning(const App& app) const {
-  bool cmd_res{false};
-  std::string cmd_output;
-  std::tie(cmd_res, cmd_output) = cmd("cat " + (appRoot(app) / ComposeFile).string());
-  if (!cmd_res) {
-    LOG_WARNING << "Failed to parse App config: " << app.name;
-    return false;
-  }
-
-  // calculate a number of container images that a given App consists of
-  std::string::size_type find_pos{0};
-  std::string::size_type line_pos{0};
-  int expected_container_number{0};
-
-  while ((find_pos = cmd_output.find("image:", line_pos)) != std::string::npos) {
-    if (cmd_output.substr(line_pos, (find_pos - line_pos)).find_first_of('#') == std::string::npos) {
-      ++expected_container_number;
+  try {
+    ComposeInfo compose((appRoot(app) / ComposeFile).string());
+
+    // get the running information from docker
+    Json::Value running;
+    ContainerClient client;
+    if (!client.getContainer(false, app.name, running)) {
```

Comment:
With such an approach, we still do one call to the docker daemon per app...

Reply:
did I misunderstand? do we not want that?
I mean it is easy to fix it - so grab all services at once and then parse.......let me know if you'd rather have that!